### PR TITLE
Ilikethese/feat/jni

### PR DESCRIPTION
**Pull request by ilikethese**
# Pre-PR Checklist

- [x] I added/updated relevant documentation.
- [x] I followed the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters to submit commit message.
- [x] I squashed the repeated code commits.
- [x] I signed the [CLA].
- [x] I added/updated test cases to check the change I am making.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=2248) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=2248) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: churchill-zhang<br/>:x: 谌立<br/><hr/>**谌立** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=2248) it.</sub>

File: driver/js/android/src/main/jni/CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +52,12 @@ GlobalPackages_Add(footstone)
 target_link_libraries(${PROJECT_NAME} PRIVATE footstone)
 # endregion
 
+# region jni
+GlobalPackages_Add(jni)
+target_link_libraries(${PROJECT_NAME} PRIVATE jni)
+target_include_directories(${PROJECT_NAME} PRIVATE $<TARGET_PROPERTY:jni,INTERFACE_INCLUDE_DIRECTORIES>)
```
Comment (medns) [inline driver/js/android/src/main/jni/CMakeLists.txt:58]: unnecessary include

File: driver/js/core/CMakeLists.txt [hunk 2]
```diff
@@ -61,6 +61,11 @@ GlobalPackages_Add(footstone)
 target_link_libraries(${PROJECT_NAME} PRIVATE footstone)
 # endregion
 
+# region jni
+GlobalPackages_Add(jni)
+target_link_libraries(${PROJECT_NAME} PRIVATE jni)
+# endregion
```
Comment (medns) [inline driver/js/core/CMakeLists.txt:67]: unnecessary import

File: modules/android/jni/CMakeLists.txt [hunk 3]
```diff
@@ -20,24 +20,32 @@
...
+project(jni)
 
 get_filename_component(PROJECT_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../.." REALPATH)
 
 include("${PROJECT_ROOT_DIR}/buildconfig/cmake/compiler_toolchain.cmake")
 
 set(CMAKE_CXX_STANDARD 17)
 
-set(ATOMIC_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+set(JNI_PATH ${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (medns) [inline modules/android/jni/CMakeLists.txt:31]: unnecessary variable

File: modules/android/jni/src/jni_env.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#pragma once
+
+#include <jni.h>
+
+#include <memory>
+#include <mutex>
+
+class JNIEnvironment {
+ public:
+  struct JNIWrapper {
```
Comment (medns) [inline modules/android/jni/src/jni_env.h:32]: struct `JNIWrapper` should NOT be here

**Review by medns — CHANGES_REQUESTED** [review]

File: driver/js/android/src/main/jni/CMakeLists.txt (hunk 1 above)
Comment (churchill-zhang) [reply driver/js/android/src/main/jni/CMakeLists.txt:58]: android的jni需要依赖外部公用jni吧

File: driver/js/core/CMakeLists.txt (hunk 2 above)
Comment (churchill-zhang) [reply driver/js/core/CMakeLists.txt:67]: runtime的耦合导致这里还需要，后续会解开

File: modules/android/jni/src/jni_env.h (hunk 4 above)
Comment (churchill-zhang) [reply modules/android/jni/src/jni_env.h:32]: 此处耦合会同runtime一起解开

File: modules/android/jni/CMakeLists.txt (hunk 3 above)
Comment (churchill-zhang) [reply modules/android/jni/CMakeLists.txt:31]: 此处可以删掉

**Pull request closed without merging**